spec/index.bs

```diff
@@ -3700,27 +3700,20 @@ GPURenderPipeline includes GPUPipelineBase;
 
 <script type=idl>
 dictionary GPURenderPipelineDescriptor : GPUPipelineDescriptorBase {
+    GPUVertexStateDescriptor vertexState = {};
     required GPUProgrammableStageDescriptor vertexStage;
-    GPUProgrammableStageDescriptor fragmentStage;
-
-    required GPUPrimitiveTopology primitiveTopology;
+    GPUPrimitiveAssemblyStateDescriptor primitiveAssemblyState = {};
     GPURasterizationStateDescriptor rasterizationState = {};
-    required sequence<GPUColorStateDescriptor> colorStates;
+    GPUProgrammableStageDescriptor fragmentStage;
     GPUDepthStencilStateDescriptor depthStencilState;
-    GPUVertexStateDescriptor vertexState = {};
-
-    GPUSize32 sampleCount = 1;
-    GPUSampleMask sampleMask = 0xFFFFFFFF;
-    boolean alphaToCoverageEnabled = false;
+    required sequence<GPUColorStateDescriptor> colorStates;
```

Comment:
Maybe this doesn't need to be required because it is valid to have a depth-only pass.

Reply:
Right, we can easily make it non-required. It's a bit strange to optimize the API for the edge cases like this though

Reply:
I think we previously decided to just make the user specify `[]` if they want that.